mongodbatlas/fw_resource_mongodbatlas_project.go

```diff
@@ -0,0 +1,857 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/setplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	expanders "github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/framework/expanders"
+)
+
+const (
+	errorProjectCreate  = "error creating Project: %s"
+	errorProjectRead    = "error getting project(%s): %s"
+	errorProjectDelete  = "error deleting project (%s): %s"
+	errorProjectSetting = "error setting `%s` for project (%s): %s"
+	errorProjectUpdate  = "error updating project (%s): %s"
+)
+
+var _ resource.Resource = &ProjectRS{}
+var _ resource.ResourceWithImportState = &ProjectRS{}
+
+func NewProjectRS() resource.Resource {
+	return &ProjectRS{}
+}
+
+type ProjectRS struct {
+	client *MongoDBClient
+}
+
+type tfProjectRSModel struct {
+	Limits                                      types.Set    `tfsdk:"limits"`
+	Teams                                       types.Set    `tfsdk:"teams"`
+	RegionUsageRestrictions                     types.String `tfsdk:"region_usage_restrictions"`
+	Name                                        types.String `tfsdk:"name"`
+	OrgID                                       types.String `tfsdk:"org_id"`
+	Created                                     types.String `tfsdk:"created"`
+	ProjectOwnerID                              types.String `tfsdk:"project_owner_id"`
+	ID                                          types.String `tfsdk:"id"`
+	ClusterCount                                types.Int64  `tfsdk:"cluster_count"`
+	IsDataExplorerEnabled                       types.Bool   `tfsdk:"is_data_explorer_enabled"`
+	IsPerformanceAdvisorEnabled                 types.Bool   `tfsdk:"is_performance_advisor_enabled"`
+	IsRealtimePerformancePanelEnabled           types.Bool   `tfsdk:"is_realtime_performance_panel_enabled"`
+	IsSchemaAdvisorEnabled                      types.Bool   `tfsdk:"is_schema_advisor_enabled"`
+	IsExtendedStorageSizesEnabled               types.Bool   `tfsdk:"is_extended_storage_sizes_enabled"`
+	IsCollectDatabaseSpecificsStatisticsEnabled types.Bool   `tfsdk:"is_collect_database_specifics_statistics_enabled"`
+	WithDefaultAlertsSettings                   types.Bool   `tfsdk:"with_default_alerts_settings"`
+}
```

Comment:
Can't we do this? Or the opposite, the DS using the resource model
```suggestion
type tfProjectRSModel struct {
tfProjectDSModel
}
```

Reply:
No. Unfortunately there were differences in all models. For example, data source models don't have `WithDefaultAlertsSettings` or `ProjectOwnerID` fields and these models need to be exactly same as the schemas

Reply:
Thanks!